### PR TITLE
feat: Account deletion command.

### PR DESCRIPTION
**Pull request by DEV-DIBSTER**
- Allows a user to delete their panel account.
- Deletes their userData.
- Deletes their userPrem.

### CODE REVIEW

**Comment by wdhdev** [comment]
Why delete premium servers?

**Comment by DEV-DIBSTER** [comment]
> Why delete premium servers?

If users at DBH only had a panel account and no other data, we can confirm that a single command ran by them would be sufficient enough for account deletion.

**Comment by wdhdev** [comment]
What if they do have premium servers? Why would we delete them?

**Comment by DEV-DIBSTER** [comment]
> What if they do have premium servers? Why would we delete them?

It deletes their account, so storing premium servers doesn't make sense in my mind.

**Comment by wdhdev** [comment]
They paid for them though. What if they want them in future?

**Comment by Microwave-6443** [comment]
> They paid for them though. What if they want them in future?

Why would you want to delete your account if you intend to use the service in the future? Especially if you've spent money on it, in this case, for premium servers.

**Comment by xrap1dx** [comment]
> Why would you want to delete your account if you intend to use the service in the future? Especially if you've spent money on it, in this case, for premium servers.

But if you ever came back to DBH then you want your premium servers? 

I feel like you should be able to use a command then it emails the panel
Account and you click confirm and within 30 days the Data gets deleted & Maybe a “Account Disable” feature?

**Comment by Microwave-6443** [comment]
> > Why would you want to delete your account if you intend to use the service in the future? Especially if you've spent money on it, in this case, for premium servers.
> 
> But if you ever came back to DBH then you want your premium servers?
> 
> I feel like you should be able to use a command then it emails the panel Account and you click confirm and within 30 days the Data gets deleted & Maybe a “Account Disable” feature?

I don't think you really succeeded in countering my point here, because again, they're likely not thinking of coming back, and that wouldn't be the host's fault. Account deletion requests are intended to delete **ALL** data related to the account, not just select pieces.

As for a 30-day confirmation period, I think that would help cement the user's decision to delete their data and it would give them plenty of time to go back if they decide that they don't actually want to delete their data.

File: src/commands/user/delete.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                });
+
+                // Delete the user's premium if they have one.
+                userData.delete(message.author.id);
+
+                // Delete the user's premium if they have one.
+                if (userPrem != null) {
+                    userPrem.delete(message.author.id);
+                }
+
```
Comment (wdhdev) [inline src/commands/user/delete.js:107]:
```suggestion
```

**Review by wdhdev — CHANGES_REQUESTED** [review]

File: src/commands/user/delete.js (hunk 1 above)
Comment (Microwave-6443) [reply src/commands/user/delete.js:107]: I guess he disagreed with us. 🤣

Comment (DEV-DIBSTER) [reply src/commands/user/delete.js:107]: 😂 Real.

**Pull request merged**